gm4_chest_boats/data/gm4_chest_boats/functions/main.mcfunction

```diff
@@ -0,0 +1,4 @@
+
+schedule function gm4_chest_boats:main 16t
+
+execute as @e[type=boat] at @s if entity @e[type=chest_minecart,distance=..2,predicate=!gm4_chest_boats:is_passenger] unless entity @p[distance=..0.5] unless data entity @s Passengers[0] run function gm4_chest_boats:merge
```

Comment:
> 
> 
> The `@e[type=boat]` should probably have a tag check to not check chest boats. The code below doesn't include that check, but improves the rest of the code.
> 
> However, I have to ask why there's a player check, when there's already a Passengers check? The header from `gm4_chest_boats:merge` indicate that it's just "no passengers on boat, no players on boat", but the passengers check will verify the players check and make it less confusing (the player could get confused if they're standing on top of the boat and the chest doesn't merge).

Currently, if you break the minecart you can just push another inside of the boat. If I added a tag then you would have to break both the boat and the minecart before recombining them which can easily lead to confusion.

When you are riding in a boat you are not listed as a passenger so it would kick you out.